FK_Project/src/Fisher1D.hpp

```diff
@@ -0,0 +1,215 @@
+#ifndef FISHER_KOLMOGOROV_HPP
+#define FISHER_KOLMOGOROV_HPP
+
+#include <deal.II/base/conditional_ostream.h>
+#include <deal.II/base/quadrature_lib.h>
+
+#include <deal.II/distributed/fully_distributed_tria.h>
+
+#include <deal.II/dofs/dof_handler.h>
+#include <deal.II/dofs/dof_tools.h>
+
+#include <deal.II/fe/fe_simplex_p.h>
+#include <deal.II/fe/fe_system.h>
+#include <deal.II/fe/fe_values.h>
+#include <deal.II/fe/fe_values_extractors.h>
+#include <deal.II/fe/mapping_fe.h>
+#include <deal.II/fe/fe_q.h>
+
+#include <deal.II/grid/grid_in.h>
+#include <deal.II/grid/grid_out.h>
+#include <deal.II/grid/tria.h>
+
+#include <deal.II/grid/grid_generator.h>
+
+
+#include <deal.II/lac/solver_cg.h>
+#include <deal.II/lac/trilinos_precondition.h>
+#include <deal.II/lac/trilinos_sparse_matrix.h>
+#include <deal.II/lac/vector.h>
+
+#include <deal.II/numerics/data_out.h>
+#include <deal.II/numerics/matrix_tools.h>
+#include <deal.II/numerics/vector_tools.h>
+
+
+#include <deal.II/lac/dynamic_sparsity_pattern.h>
+#include <deal.II/lac/precondition.h>
+#include <deal.II/lac/sparse_matrix.h>
+#include <deal.II/lac/vector.h>
+
+#include <fstream>
+#include <iostream>
+
+using namespace dealii;
+
+class Fisher1D
+{
+public:
+
+    // Physical dimension: for this case, dim = 1
+    static constexpr unsigned int dim = 1;
+
+    // Function describing the value of the spreading coefficient -> the bigger alpha is, the faster is the spreading
+    class FunctionAlpha : public Function<dim> 
+    {
+    public : 
+        virtual double 
+        value(const Point<dim> & /*p*/,
+            const unsigned int /*component*/ = 0) const /*override*/
+        {   
+            return 2.0;
+        }
+    }; 
+
+    // Function describing the behaviour of D
+    class FunctionD : public Function<dim> 
+    {
+    public : 
+        virtual double 
+        value(const Point<dim> & /*p*/,
+            const unsigned int /*component*/ = 0) const /*override*/
+        {   
+            return 0.0004;
+        }
+
+    }; 
+
+    // Function for the initial concentration in a specific region of the mesh
+    class FunctionC0 : public Function<dim>
+    {
+    public:
+        virtual double
+        value(const Point<dim> &p,
+            const unsigned int /*component*/ = 0) const /*override*/
+        {
+            //Point<dim> origin;
+            Point<dim> starting_point(0.0);
+            double max_value = 0.1;
+            double std_dev = 0.0333;
```

Comment:
```suggestion
            const Point<dim> starting_point(0.0);
            constexpr double max_value = 0.1;
            constexpr double std_dev = 0.0333;
```
Declaring variables as `const` or `constexpr` greatly facilitates compiler optimization (e.g. the compiler would likely precompute `2 * std_dev * std_dev` if `std_dev` was declared as `constexpr`, and it might do so even if it was declared as `const`). Additionally, these could more appropriately be defined as members of the class, rather than local variables to the `value` method.